### ISSUE TITLE
Does web-ext disable e10s?

### ISSUE THREAD

**Issue by andymckay** [mozilla/web-ext#970]
### Is this a feature request or a bug?

Bug

### What is the current behavior?

For the moment don't use web-ext. Get a Firefox with a clean profile and make sure e10s is enabled in about:support. Try out this add-on: https://gist.github.com/kentbrew/190a0ad83f037567eb2393c235e78043. Go to example.com, click on pinterest in the frame. Notice it won't open the pinterest page.

Ok, so that this bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1357589, it doesn't work with e10s on.

Now go try this with web-ext. It works and opens a pinterest page.

Why's that then? Has web-ext disabled e10s somehow? Or is something else going on.

### What is the expected or desired behavior?

web-ext should be as close to release as possible.

### Version information (for bug reports)

* **Firefox version**: all
* **Your OS and version**: os x
* Paste the output of these commands:

````
node --version && npm --version && web-ext --version
````

v7.8.0
4.2.0
1.9.1

/cc @rpl who likely knows the most about this.

**Comment by andymckay** [comment]
It's not e10s, its because: "Open new windows in a new tab instead" in about:preferences is not turned on. On a new Firefox Nightly it is.

**Comment by kentbrew** [comment]
When using `web-ext` instead of loading through `about:debugging`:

This extension opens the new page in a new window but gets me blank tabs when I turn on "Open new windows in a new tab instead"

**Comment by kentbrew** [comment]
I've just gone through the process of building and signing the test add-on and then installing it as a permanent extension. It is behaving the same way; when "Open new windows in a new tab instead" is selected in about:preferences, we get an empty tab.

**Comment by andymckay** [comment]
That's bug https://bugzilla.mozilla.org/show_bug.cgi?id=1357589. I'd like to just have web-ext be as close to the default experience as possible.

**Comment by andymckay** [comment]
That's bug https://bugzilla.mozilla.org/show_bug.cgi?id=1357589. I'd like to just have web-ext be as close to the default experience as possible.

**Comment by kumar303** [comment]
Maybe it's one of the prefs we set? https://github.com/mozilla/web-ext/blob/master/src/firefox/preferences.js#L22

**Comment by rpl** [comment]
@andymckay @kumar303 I think that this is likely to be related to the Firefox preferences that are implicitly set internally by the firefox-profile-js dependency, in particular this line:

- https://github.com/saadtazi/firefox-profile-js/blob/b5821e92f7614c6f8ff6a4a0c93ff5264990ddb3/lib/firefox_profile.js#L32 

The `browser.link.open_newwindow` preference is explicitly set to 2 by firefox-profile-js, while its default is 3 in mozilla-central:

- http://searchfox.org/mozilla-central/rev/d840ebd5858a61dbc1622487c1fab74ecf235e03/browser/app/profile/firefox.js#424

**Comment by apoorvasingh17** [comment]
Hi, can I work on this?

**Comment by kumar303** [comment]
@apoorvasingh17 I don't think this can be fixed in the web-ext codebase if I'm reading https://github.com/mozilla/web-ext/issues/970#issuecomment-307893927 correctly

**Comment by joshhus** [comment]
Still no <a target=new href= ... > or {:new_window} functionality available in GitHub repos, to open link in a 2nd browser tab/window?
